stdlib/regex.gr

```diff
@@ -0,0 +1,2983 @@
+    numGroups: Array.length(state) + 1,
+    allGroupPositions: getAllMatchGroupPositions,
+    allGroups: getAllMatchGroups
+  }
+}
+
+// Helpers for user-facing match functionality
+
+let fastDriveRegexIsMatch = (rx, string, startOffset, endOffset) => {
+  let state = if (rx.reReferences) Array.make(rx.reNumGroups, None) else Array.make(0, None)
+  let toWrap = if (startOffset == 0 && endOffset == String.length(string)) string else String.slice(startOffset, endOffset, string)
+  let buf = makeMatchBuffer(toWrap)
+  Option.isSome(searchMatch(rx, buf, 0, 0, Array.length(buf.matchInputExploded), state))
+}
+
+let rec fastDriveRegexMatchAll = (rx, string, startOffset, endOffset) => {
+  if (startOffset >= endOffset) {
+    []
+  } else {
+    let state = Array.make(rx.reNumGroups, None)
+    let toWrap = if (startOffset == 0 && endOffset == String.length(string)) string else String.slice(startOffset, endOffset, string)
+    let buf = makeMatchBuffer(toWrap)
+    match(searchMatch(rx, buf, 0, 0, Array.length(buf.matchInputExploded), state)) {
+      None => [],
+      Some((startPos, endPos)) => [makeMatchResult(string, startPos + startOffset, endPos + startOffset, Array.map(elt => {
+        match(elt) {
+          None => None,
+          Some((start, end)) => Some((start + startOffset, end + startOffset))
+        }
+      }, state)), ...fastDriveRegexMatchAll(rx, string, startPos + startOffset + 1, endOffset)],
+    }
+  }
+}
+
+let fastDriveRegexMatch = (rx, string, startOffset, endOffset) => {
+  let state = Array.make(rx.reNumGroups, None)
+  let toWrap = if (startOffset == 0 && endOffset == String.length(string)) string else String.slice(startOffset, endOffset, string)
+  let buf = makeMatchBuffer(toWrap)
+  match(searchMatch(rx, buf, 0, 0, Array.length(buf.matchInputExploded), state)) {
+    None => None,
+    Some((startPos, endPos)) => {
+      Some(makeMatchResult(string, startPos + startOffset, endPos + startOffset, Array.map(elt => {
+        match(elt) {
+          None => None,
+          Some((start, end)) => Some((start + startOffset, end + startOffset))
+        }
+      }, state)))
+    }
+  }
+}
+
+/**
+ * Returns true if the given regular expression has a match in the given string
+ * @param rx: RegularExpression - The regular expression to search for
+ * @param string: String - The string to search within
+ * @returns Whether `rx` matches `string`
+ */
+export let isMatch = (rx: RegularExpression, string: String) => {
+  fastDriveRegexIsMatch(rx, string, 0, String.length(string))
+}
+
+/**
+ * Returns true if the given regular expression has a match in the given string within the given start/end offsets
+ * @param rx: RegularExpression - The regular expression to search for
+ * @param string: String - The string to search within
+ * @param start: Number - The start offset to search within
+ * @param end: Number - The end offset to search within
+ * @returns Whether `rx` matches `string` in the given range
```

Comment:
```suggestion
 * Determines if the given regular expression has a match in the given string between the given start/end offsets.
 * @param rx: The regular expression to search for
 * @param string: The string to search
 * @param start: The start offset to search between
 * @param end: The end offset to search between
 * @returns `true` if the RegExp matches the string in the given range, otherwise `false`
```